perception/autoware_tensorrt_bevdet/include/autoware/tensorrt_bevdet/bevpool_plugin.hpp

```diff
@@ -0,0 +1,123 @@
+ * limitations under the License.
+ */
+#ifndef AUTOWARE__TENSORRT_BEVDET__BEVPOOL_PLUGIN_HPP_
+#define AUTOWARE__TENSORRT_BEVDET__BEVPOOL_PLUGIN_HPP_
+
+#include <NvInfer.h>
+#include <unistd.h>
+
+#include <cmath>
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+namespace
+{
+static const char * BEVPOOL_PLUGIN_NAME{"BEVPool"};
+static const char * BEVPOOL_PLUGIN_VERSION{"1"};
+}  // namespace
+
+namespace nvinfer1
+{
+class BEVPoolPlugin : public IPluginV2DynamicExt
+{
+private:
+  const std::string name_;
+  std::string namespace_;
+  struct
+  {
+    int bev_h;
+    int bev_w;
+    int n;
+  } m_;
+
+public:
+  BEVPoolPlugin() = delete;
+  BEVPoolPlugin(const std::string & name, int bev_h, int bev_w, int n);
+  BEVPoolPlugin(const std::string & name, const void * buffer, size_t length);
+  ~BEVPoolPlugin();
+
+  // Method inherited from IPluginV2
+  const char * getPluginType() const noexcept override;
+  const char * getPluginVersion() const noexcept override;
+  int32_t getNbOutputs() const noexcept override;
+  int32_t initialize() noexcept override;
+  void terminate() noexcept override;
+  size_t getSerializationSize() const noexcept override;
+  void serialize(void * buffer) const noexcept override;
+  void destroy() noexcept override;
+  void setPluginNamespace(const char * pluginNamespace) noexcept override;
+  const char * getPluginNamespace() const noexcept override;
+
+  // Method inherited from IPluginV2Ext
+  DataType getOutputDataType(
+    int32_t index, DataType const * inputTypes, int32_t nbInputs) const noexcept override;
+  void attachToContext(
+    cudnnContext * contextCudnn, cublasContext * contextCublas,
+    IGpuAllocator * gpuAllocator) noexcept override;
+  void detachFromContext() noexcept override;
+
+  // Method inherited from IPluginV2DynamicExt
+  IPluginV2DynamicExt * clone() const noexcept override;
+  DimsExprs getOutputDimensions(
+    int32_t outputIndex, const DimsExprs * inputs, int32_t nbInputs,
+    IExprBuilder & exprBuilder) noexcept override;
+  bool supportsFormatCombination(
+    int32_t pos, const PluginTensorDesc * inOut, int32_t nbInputs,
+    int32_t nbOutputs) noexcept override;
+  void configurePlugin(
+    const DynamicPluginTensorDesc * in, int32_t nbInputs, const DynamicPluginTensorDesc * out,
+    int32_t nbOutputs) noexcept override;
+  size_t getWorkspaceSize(
+    const PluginTensorDesc * inputs, int32_t nbInputs, const PluginTensorDesc * outputs,
+    int32_t nbOutputs) const noexcept override;
+  int32_t enqueue(
+    const PluginTensorDesc * inputDesc, const PluginTensorDesc * outputDesc,
+    const void * const * inputs, void * const * outputs, void * workspace,
+    cudaStream_t stream) noexcept override;
+
+protected:
+  // To prevent compiler warnings
+  using nvinfer1::IPluginV2::enqueue;
+  using nvinfer1::IPluginV2::getOutputDimensions;
+  using nvinfer1::IPluginV2::getWorkspaceSize;
+  using nvinfer1::IPluginV2Ext::configurePlugin;
+};
+
```

Comment:
Similarly, the anonymous namespace is unnecessary here. The entire class definition should be placed within the `nvinfer1` namespace to maintain consistency and avoid confusion.